### PR TITLE
remove to_regex dependency and the warnings it causes

### PR DESCRIPTION
**Pull request by grosser**
`vendor/bundle/ruby/2.6.0/gems/to_regexp-0.2.1/lib/to_regexp.rb:35: warning: `*' interpreted as argument prefix`

fixes https://github.com/joshbuddy/jsonpath/issues/125

### CODE REVIEW

File: jsonpath.gemspec [hunk 1]
```diff
@@ -5,28 +5,20 @@ require File.join(File.dirname(__FILE__), 'lib', 'jsonpath', 'version')
...
-  end
+  s.required_rubygems_version = Gem::Requirement.new('>= 0')
   s.authors = ['Joshua Hull', 'Gergely Brautigam']
   s.summary = 'Ruby implementation of http://goessner.net/articles/JsonPath/'
   s.description = 'Ruby implementation of http://goessner.net/articles/JsonPath/.'
   s.email = ['joshbuddy@gmail.com', 'skarlso777@gmail.com']
   s.extra_rdoc_files = ['README.md']
   s.files = `git ls-files`.split("\n")
   s.homepage = 'https://github.com/joshbuddy/jsonpath'
-  s.rdoc_options = ['--charset=UTF-8']
```
Comment (Skarlso) [inline jsonpath.gemspec:19]: What was the reason for removing these options?

**Comment by grosser** [comment]
it's the default anyway

**Comment by Skarlso** [comment]
So I'm wondering if the test failures are related or not. 🤔

**Comment by grosser** [comment]
clean PR https://github.com/joshbuddy/jsonpath/pull/129

**Comment by grosser** [comment]
and https://github.com/joshbuddy/jsonpath/pull/130 while we are at it ...

**Comment by grosser** [comment]
same error on master ... so merge this and then fix ?

**Comment by Skarlso** [comment]
I'll take a look at this tonight and fix the build then merge stuff. :) I don't trust a broken system. :)

**Comment by Skarlso** [comment]
@grosser Please merge master and run this again

**Comment by grosser** [comment]
jruby still unhappy ... sane as master

**Review by Skarlso — APPROVED** [review]
LGTM

File: lib/jsonpath/parser.rb [hunk 2]
```diff
@@ -102,6 +106,31 @@ def parse_exp(exp)
 
     private
 
+    # /foo/i -> Regex.new("foo", Regexp::IGNORECASE) without using eval
+    # also supports %r{foo}i
```
Comment (Skarlso) [inline lib/jsonpath/parser.rb:110]: @grosser Would you mind adding some test case for these options please to see if they are even usable and don't break things unexpectedly?

**Comment by Skarlso** [comment]
I'll add the test later.

Jruby has been identified: https://github.com/jruby/jruby/issues/6469

**Pull request merged**